io/plugins/eu.esdihumboldt.hale.io.appschema.test/src/eu/esdihumboldt/hale/io/appschema/writer/AppSchemaMappingTest.java

```diff
@@ -336,30 +363,179 @@ public void testXrefJoinHandler() {
 		assertNull(nestedMapping.getSourceExpression().getLinkField());
 	}
 
-	private DefaultCell buildJoinCell() {
+	@Test
+	public void testFeatureChainingJoin() throws Exception {
+		FeatureChaining chainingConf = readFeatureChainingConfiguration(FEATURE_CHAINING_CONF_PATH);
+		assertNotNull(chainingConf);
+
+		DefaultCell joinCell = buildJoinCell(chainingConf);
+
+		// create minimal alignment and pass it to JoinHandler
+		DefaultAlignment alignment = new DefaultAlignment();
+		alignment.addCell(joinCell);
+
+		DefaultCell renameCell = new DefaultCell();
+		renameCell.setTransformationIdentifier(RenameFunction.ID);
+		renameCell.setSource(getUuidSourceProperty(unitType));
+		renameCell.setTarget(getNestedUnitLocalIdTargetProperty());
+		alignment.addCell(renameCell);
+
+		DefaultCell renameClassCell = new DefaultCell();
+		renameClassCell.setTransformationIdentifier(RenameFunction.ID);
+		renameClassCell.setSource(getObservationClassSourceProperty());
+		renameClassCell.setTarget(getNestedObservationClassTargetProperty());
+		alignment.addCell(renameClassCell);
+
+//		processJoinAlignment(alignment, chainingConf);
```

Comment:
I also forgot to comment out (or remove) the two `logMapping` calls in the following lines... could you do it for me, please?